src/Query/Builder.php

```diff
@@ -732,6 +733,11 @@ public function upsert(array $values, $uniqueBy, $update = null): int
             return 0;
         }
 
+        // Single document provided
+        if (! is_array(reset($values))) {
```

Comment:
Just double-checking that we never expect this method to be called with an array of objects? Technically the driver would be able to store those.

Reply:
This was copied from Laravel: https://github.com/laravel/framework/blob/d16ce251beb53f46d602f5f523ba82900ae4be91/src/Illuminate/Database/Eloquent/Builder.php#L1132-L1135

But I could use [`array_is_list`](https://www.php.net/manual/en/function.array-is-list.php) to check the array instead of the values.

```suggestion
        if (! array_is_list($values)) {
```

Reply:
I think that's a better option in general, regardless of whether or not the method should accept objects.